packages/open-next/src/adapters/cache.ts

```diff
@@ -160,96 +177,59 @@ export default class S3Cache {
+        LastModified?.getTime(),
+      );
+      if (lastModified === -1) {
+        // If some tags are stale we need to force revalidation
+        return null;
+      }
+      if (cacheData.type === "route") {
         return {
           lastModified: LastModified?.getTime(),
           value: {
             kind: "ROUTE",
-            body: Buffer.from(body ?? Buffer.alloc(0)),
-            status: meta.status,
-            headers: meta.headers,
+            body: Buffer.from(cacheData.body ?? Buffer.alloc(0)),
+            status: meta?.status,
+            headers: meta?.headers,
           },
         } as CacheHandlerValue;
-      } catch (e) {
-        error("Failed to get body cache", e);
-      }
-      return null;
-    }
-
-    if (keys.includes(this.buildS3Key(key, "html"))) {
-      const isJson = keys.includes(this.buildS3Key(key, "json"));
-      const isRsc = keys.includes(this.buildS3Key(key, "rsc"));
-      debug("get html cache ", { key, isJson, isRsc });
-      if (!isJson && !isRsc) return null;
-
-      try {
-        const [{ Body, LastModified }, { Body: PageBody }, { Body: MetaBody }] =
-          await Promise.all([
-            this.getS3Object(key, "html", keys),
-            this.getS3Object(key, isJson ? "json" : "rsc", keys),
-            this.getS3Object(key, "meta", keys),
-          ]);
-        const lastModified = await this.getHasRevalidatedTags(
-          key,
-          LastModified?.getTime(),
-        );
-        if (lastModified === -1) {
-          // If some tags are stale we need to force revalidation
-          return null;
-        }
-        const meta = JSON.parse((await MetaBody?.transformToString()) ?? "{}");
+      } else if (cacheData.type === "page" || cacheData.type === "app") {
         return {
-          lastModified,
+          lastModified: LastModified?.getTime(),
           value: {
             kind: "PAGE",
-            html: (await Body?.transformToString()) ?? "",
-            pageData: isJson
-              ? JSON.parse((await PageBody?.transformToString()) ?? "{}")
-              : await PageBody?.transformToString(),
-            status: meta.status,
-            headers: meta.headers,
+            html: cacheData.html,
+            pageData:
+              cacheData.type === "page" ? cacheData.json : cacheData.rsc,
+            status: meta?.status,
+            headers: meta?.headers,
           },
         } as CacheHandlerValue;
-      } catch (e) {
-        error("Failed to get html cache", e);
-      }
-      return null;
-    }
-
-    // Check for redirect last. This way if a page has been regenerated
-    // after having been redirected, we'll get the page data
-    if (keys.includes(this.buildS3Key(key, "redirect"))) {
-      debug("get redirect cache", { key });
-      try {
-        const { Body, LastModified } = await this.getS3Object(
-          key,
-          "redirect",
-          keys,
-        );
+      } else if (cacheData.type === "redirect") {
```

Comment:
Is there a case for `rewrites`?

Reply:
No this is only for `redirect` inside `getStaticProps` or inside an ISR or SSG app route